Importer Basics

jonuy opened the pull request:
#### What's this PR do?

Provides foundation for importing the SMS game json config files and converting them into Twine stories. Currently only supports the story config passage and regular DS passages.

Also some cleanup with the project name and fields in the package.json.
#### Where should the reviewer start?
###### storylistview.js

The file being imported is read here. I added pulling the filename and extension in order to know if it should be processed as one of our .json config files instead.
###### app.js

Basically just acts as a bridge to importsms. Seems like the only reason I have this is to stay consistent with how the original `app.importFile` worked.
###### importsms.js

Reads through all a config object's keys and creates corresponding passages to display in Twine.
#### How should this be manually tested?

I tested with this file: https://gist.github.com/jonuy/88a949644e3eb437da0e

Import it using the "Import From File" button on the Intertwine homepage. Verify that 4 passages get created with data and are connected.

@tongxiang

jonuy committed:
Added importsms lib and modified 'Import From File' to handle sms game .json config files
- importer currently only handles story config and PassageDS models

jonuy committed:
Changed name and package.json info

jonuy committed:
Changing key regex testing from using match() to test()
- only needed a boolean answer in these cases, not an array

jonuy committed:
Creating passages from arrays instead of one object at a time
- in preparation of needing to do this for end-level passages

tongxiang commented on js/app.js:
```diff
@@ -17,7 +17,7 @@ var TwineApp = Backbone.Marionette.Application.extend(
   **/
 
-  name: 'Twine',
+  name: 'Intertwine',
```
:santa: We made it!!!! :octopus:

tongxiang commented on js/importsms.js:
```diff
@@ -0,0 +1,164 @@
+
+    data = JSON.parse(strData);
+
+    allStories = StoryCollection.all();
```
Why is it necessary to run the `StoryCollection.all()` function to create stories?

jonuy replied:
Good question. I have no idea. I just tried it with `new StoryCollection()` and `new PassageCollection()` instead and it seems to work. I'll submit a change in a sec.

jonuy committed:
Using new PassageCollection and StoryCollection objects on the import
- mostly to remove any confusion. It worked just as fine before as it does now, but it was unclear why we did a fetch on the collections there (and why it happens in app.importFile)

tongxiang commented:
:+1:
